test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -138,10 +138,13 @@ describe('mainTemplate.json file - ', () => {
                     var val = templateObject.resources[resource];
                     if (val.type) {
                         var rType = val.type.split('/');
+                        var lowerType = val.type.toLowerCase();
```

Comment:
Can we make this consistent across the checks?  i.e. use rType[0].toLowerCase() or lowerType but not both?

Reply:
There are two class of tests here: anything from an RP (`Microsoft.Compute/*`), and specific resource type (`Microsoft.Insights/autoscaleSettings`).

I could change the new tests to check `rType[0]` and `rType[1]`? Or convert the existing `rType[0]` check into `lowerType.startsWith('microsoft.compute/')`?

Thoughts?

Reply:
Or, how about having a data structure with a specification of the allowed types and have the condition look up in that? Might make it clearer going forwards as more types are added?

E.g. 
```javascript
            var allowedResourceTypes = {
                'microsoft.compute' : ['*'],
                'microsoft.storage' : ['*'],
                'microsoft.network' : ['*'],
                'microsoft.resources' : ['*'],
                'microsoft.insights' : ['autoscalesettings'],
                'microsoft.authorization' : ['roleassignments']
            };
```